src/models/shape_loader.cpp

```diff
@@ -732,20 +738,29 @@ geo::ShapePtr loadShape(const std::string& model_path, tue::config::Reader cfg,
         }
 
         readPose(cfg, pose);
+        cfg.endGroup();
     }
     else if (cfg.readGroup("cylinder")) // SDF AND ED YAML
     {
         int num_points = 12;
         cfg.value("num_points", num_points, tue::config::OPTIONAL);
 
         double radius, height;
-        if (cfg.value("radius", radius) && (cfg.value("height", height) || cfg.value("length", height))) //length is used in SDF
+        if (cfg.value("radius", radius) && cfg.value("height", height)) //length is used in SDF
         {
+            cfg.endGroup();
+            readPose(cfg, pose);
```

Comment:
This pose is used in the end. If in the end no pose is read(readPose is trying, but doesn't find something to read), this pose is used. To my knowledge max one of the two pose are present. Because reading one in the end would override the first one. To be 100% robust, we could create an extra pose. And check on the product of the two poses.

Reply:
But in line 915 we read the pose again. So we perform readpose twice without using the result. This operation is performed at the same level of cfg. So what exactly went wrong in the previous version of this piece of code (apart from the missing cfg.endGroup())

Reply:
This was indeed not an improvement, so I reverted this part of the changes.